qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -0,0 +1,861 @@
+    This mechanism also allows us to easily share a pulse program including
+    custom pulse instructions with collaborators.
+
+    .. note::
+
+        Currently QPY serialization of :class:`SymbolicPulse` is not available.
+        This feature will be implemented shortly.
+    """
+
+    __slots__ = (
+        "_pulse_type",
+        "_params",
+        "_envelope",
+        "_constraints",
+        "_valid_amp_conditions",
+    )
+
+    # Lambdify caches keyed on sympy expressions. Returns the corresponding callable.
+    _envelope_lam = LambdifiedExpression("_envelope")
+    _constraints_lam = LambdifiedExpression("_constraints")
+    _valid_amp_conditions_lam = LambdifiedExpression("_valid_amp_conditions")
+
+    def __init__(
+        self,
+        pulse_type: str,
+        duration: Union[ParameterExpression, int],
+        parameters: Optional[Dict[str, Union[ParameterExpression, complex]]] = None,
+        name: Optional[str] = None,
+        limit_amplitude: Optional[bool] = None,
+        envelope: Optional[sym.Expr] = None,
+        constraints: Optional[sym.Expr] = None,
+        valid_amp_conditions: Optional[sym.Expr] = None,
+    ):
+        """Create a parametric pulse.
+
+        Args:
+            pulse_type: Display name of this pulse shape.
+            duration: Duration of pulse.
+            parameters: Dictionary of pulse parameters that defines the pulse envelope.
+            name: Display name for this particular pulse envelope.
+            limit_amplitude: If ``True``, then limit the absolute value of the amplitude of the
+                waveform to 1. The default is ``True`` and the amplitude is constrained to 1.
+            envelope: Pulse envelope expression.
+            constraints: Pulse parameter constraint expression.
+            valid_amp_conditions: Extra conditions to skip a full-waveform check for the
+                amplitude limit. If this condition is not met, then the validation routine
+                will investigate the full-waveform and raise an error when the amplitude norm
+                of any data point exceeds 1.0. If not provided, the validation always
+                creates a full-waveform.
+
+        Raises:
+            PulseError: When not all parameters are listed in the attribute :attr:`PARAM_DEF`.
+        """
+        super().__init__(
+            duration=duration,
+            name=name,
+            limit_amplitude=limit_amplitude,
+        )
+        if parameters is None:
+            parameters = {}
+
+        # TODO remove this.
+        #  This is due to convention in IBM Quantum backends where "amp" is treated as a
+        #  special parameter that must be defined in the form [real, imaginary].
+        #  this check must be removed because Qiskit pulse should be backend agnostic.
+        if "amp" in parameters and not isinstance(parameters["amp"], ParameterExpression):
+            parameters["amp"] = complex(parameters["amp"])
```

Comment:
Is there a time frame or preconditions for the removal here? The reasoning is outlined on why we want to remove it, but not when we can.

Reply:
@wshanks any plan for this? Anyways updating this doesn't break anything.

Reply:
I would like to see it done, but it is hard to put a time frame on it because I can't prioritize it over things right now and I am not sure who else would be interested in implementing the change in the IBM backend.